File: src/pocketcloud/cloud/network/packet/impl/normal/ModuleSyncPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\normal;

use pocketcloud\cloud\cache\InGameModule;
use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\PacketData;

class ModuleSyncPacket extends CloudPacket {

    private array $data;

    public function __construct() {
        $this->data = [
            "sign" => ["enabled" => InGameModule::getModuleState(InGameModule::SIGN_MODULE)],
            "npc" => ["enabled" => InGameModule::getModuleState(InGameModule::NPC_MODULE)],
            "hub_command" => ["enabled" => InGameModule::getModuleState(InGameModule::HUB_COMMAND_MODULE)],
        ];
    }

    public function encodePayload(PacketData $packetData): void {
        $packetData->write($this->data);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->data = $packetData->readArray();
    }

    public function getData(): array {
        return $this->data;
    }

    public function handle(ServerClient $client): void {}

    public static function create(): self {
        return new self();
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/ModuleSyncPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Collection;
import java.util.List;

@NoArgsConstructor
@Getter
public final class ModuleSyncPacket extends CloudPacket implements ClientboundPacket {

    private Collection<String> enabledModules;

    public ModuleSyncPacket(Collection<String> enabledModules) {
        this.enabledModules = enabledModules != null ? enabledModules : List.of();
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(enabledModules);
    }

    @Override
    public void decodePayload(IPacketData packetData) {}

    public static ModuleSyncPacket create(Collection<String> enabledModules) {
        return new ModuleSyncPacket(enabledModules);
    }
}